.github/workflows/release.yml

```diff
@@ -0,0 +1,248 @@
+    #needs: [initialize, build_backend, build_frontend, integration_tests_backend, integration_tests_frontend]
+    needs: [initialize, build_backend, build_frontend]
+    env:
+      RELEASE_TYPE: ${{ needs.initialize.outputs.release_type }}
+      RELEASE_VERSION: ${{ needs.initialize.outputs.release_version }}
+      BRANCH_VERSION: ${{ needs.initialize.outputs.branch_version }}
+      NEXT_VERSION: ${{ needs.initialize.outputs.next_version }}      
+      RELEASE_BRANCH: ${{ github.event.inputs.release_branch }}
+      QUAY_TAG: ${{ github.event.inputs.quay_repository }}:${{ needs.initialize.outputs.release_version }}      
+      GH_PULL_URI: https://api.github.com/repos/${{ github.event.inputs.repository }}/pulls
+      GH_RELEASE_URI: https://api.github.com/repos/${{ github.event.inputs.repository }}/releases
+    steps:    
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.event.inputs.repository }}
+          ref: ${{ github.event.inputs.release_branch }} 
+      
+      - name: Set version to release
+        run: sed -i -r 's/^VERSION \?= v.*/VERSION \?= $RELEASE_VERSION/' Makefile        
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.7
+
+      - name: Cache Go deps
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Download frontend build
+        uses: actions/download-artifact@v3
+        with:
+          name: build
+          path: frontend/build       
+
+      - name: Build image     
+        run: make clean build 
+
+      - name: Push image
+        run: |
+          docker login -u ${{ secrets.QUAY_USER }} -p ${{ secrets.QUAY_PASSWORD }} quay.io
+        
+          make -e DOCKER_CLI_EXPERIMENTAL=enabled build-linux-multi-arch container-multi-arch-push-kiali-quay
+
+      - name: Configure git
+        run: |
+          git config user.email 'kiali-dev@googlegroups.com'
+        
+          git config user.name 'kiali-bot'
+
+      - name: Create tag 
+        if: ${{ needs.initialize.outputs.release_type != 'edge' }}
+        run: |
+          git add Makefile
+        
+          git commit -m "Release $RELEASE_VERSION"
+        
+          git push origin $(git rev-parse HEAD):refs/tags/$RELEASE_VERSION
+        
+      - name: Create release
+        run: |
+          if [[ $RELEASE_TYPE == *"snapshot"* ]]; then export PRERELEASE="true"; else export PRERELEASE="false"; fi
+
+          curl -H "Authorization: token ${{ secrets.GH_TOKEN }}" \
+            -H "Content-Type: application/json" \
+            -d "{\"name\": \"Kiali Operator $RELEASE_VERSION\", \"tag_name\": \"$RELEASE_VERSION\", \"prerelease\": $PRERELEASE}" \
+            -X POST $GH_RELEASE_URI        
```

Comment:
I'd suggest to use an action rather than invoking the API with curl.
Here is the official but deprecated action: https://github.com/actions/create-release. It is deprecated, but it lists a few alternate ones that are still maintained.